### PR TITLE
llvm-wrapper: remove llvm 12 hack

### PR DESCRIPTION
**Pull request by klensy**
effectively reverts https://github.com/rust-lang/rust/commit/9a8acea78355b604dbeb29bc38bd4dbf7bfce95f

r? @nikic

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
@bors r-
probably caused this llvm test failure? https://github.com/rust-lang/rust/pull/120394#issuecomment-1912761455

**Comment by nikic** [comment]
Yes, that's how failure to run NameAnonGlobals manifests.

File: compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp [hunk 1]
```diff
@@ -945,12 +945,7 @@ LLVMRustOptimize(
         break;
       case LLVMRustOptStage::PreLinkThinLTO:
         MPM = PB.buildThinLTOPreLinkDefaultPipeline(OptLevel);
-        // The ThinLTOPreLink pipeline already includes ThinLTOBuffer passes. However, callback
-        // passes may still run afterwards. This means we need to run the buffer passes again.
-        // FIXME: In LLVM 13, the ThinLTOPreLink pipeline also runs OptimizerLastEPCallbacks
-        // before the RequiredLTOPreLinkPasses, in which case we can remove these hacks.
-        if (OptimizerLastEPCallbacks.empty())
-          NeedThinLTOBufferPasses = false;
+        NeedThinLTOBufferPasses = false;
```
Comment (nikic) [inline compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp:948]: I believe you also need to drop the `OptStage != LLVMRustOptStage::PreLinkThinLTO` condition above and drop the manual OptimizerLastEPCallbacks scheduling below.

Comment (klensy) [reply compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp:948]: Weird, locally `tests/ui/sanitize/new-llvm-pass-manager-thin-lto.rs` passes for me on `x86_64-pc-windows-msvc`.

**Pull request merged**